src/nupic/research/TP_shim.py

```diff
@@ -101,11 +101,12 @@ def compute(self, bottomUpInput, enableLearn, computeInfOutput=None):
     numberOfCells = self.numberOfCells()
 
     activeState = numpy.zeros(numberOfCells)
-    activeState[self.getCellIndices(self.activeCells)] = 1
+    activeState[self.getActiveCells()] = 1
     self.infActiveState["t"] = activeState
 
     output = numpy.zeros(numberOfCells)
-    output[self.getCellIndices(self.predictiveCells | self.activeCells)] = 1
+    output[list(set(self.getPredictiveCells()) | 
+                set(self.getActiveCells()))] = 1
```

Comment:
It should be possible to just do:

    output[self.getPredictiveCells() + self.getActiveCells()] = 1